### PR TITLE
Generalize I-SUPPORT rigid and pneumatic segment layouts

### LINKED ISSUE #112: [AM-ISupport] Add Density and Radius for Rigid Link
I added the rigid link density and radius to both the `Params` and `ISupport` classes in this [branch](https://github.com/tud-phi/soromox/tree/rigid_density_radius). However, the rigid link's cross-sectional area and second moment of area are currently inherited from the soft segment geometry. Since in the AM-I Support, the soft segment has an annular cross-section while the rigid interfaces are cylindrical, these properties should be decoupled.

### PR DESCRIPTION
**Pull request by mstoelzle**
## Motivation

The I-SUPPORT model previously treated rigid interfaces as optional slots around a sequence of pneumatic segments. That representation assumed a fixed connector/pneumatic alternation, prevented adjacent segments of the same type, and forced rigid lengths into a separate `rigid_connector_lengths` field. It also meant that rigid radius, density, material fields, geometry, and rendering were not consistently described by the model's ordinary per-segment parameters.

This PR makes the physical robot layout explicit. Every modeled segment now occupies one entry in the standard PCS body arrays, while pneumatic-only chamber and subdivision data remain indexed by pneumatic segment order.

## What changed

- Replace `rigid_connector_selector` with `rigid_segment_selector`.
  - It contains one boolean per physical segment.
  - `True` means rigid and `False` means pneumatic.
  - When omitted, the selector defaults to `[rigid, pneumatic, rigid, pneumatic, ...]`.
  - Arbitrary layouts are supported, including adjacent rigid or pneumatic segments and layouts beginning or ending with either type.
- Remove `rigid_connector_lengths` and the canonical connector-slot abstraction.
- Index `length`, `radius`, `density`, elastic moduli, material damping, reference strain, and custom damping matrices by physical segment.
- Keep chamber geometry, `pcs_segment_counts`, pressure inputs, and subdivision groups indexed only by pneumatic segments in their order of appearance.
- Validate selector dimensions, body and chamber array shapes, at least one pneumatic segment, subdivision sums, and straight rigid reference strains.
- Expand pneumatic physical segments into their child PCS segments while preserving explicitly supplied rigid properties and pressure-group ordering.
- Use filled-cylinder geometry for rigid segments:
  - `A = pi r^2`
  - `Ixx = pi r^4 / 2`
  - `Iyy = Izz = pi r^4 / 4`
- Continue using the actuator-annulus and parallel-axis geometry for pneumatic segments.
- Update the specialized Viser renderer to traverse the generalized physical layout and render every rigid segment with its own modeled radius, including the final rigid segment.
- Update the simulation example with physical-layout arrays and values recovered from the compiled reference model: 1210 kg/m^3 for rigid interfaces and 1104 kg/m^3 for pneumatic segments.
- Standardize documentation and comments on the term “pneumatic segment”.

## New interface

All standard PCS body fields describe physical segments in base-to-tip order:

```python
import jax.numpy as jnp

from soromox.systems import ISupport, ISupportParams, ISupportStructure

# Physical layout:
# rigid base -> pneumatic -> rigid interface -> pneumatic -> rigid tip
rigid_segment_selector = (True, False, True, False, True)

params = ISupportParams(
    length=jnp.array([0.041, 0.190, 0.027, 0.180, 0.006]),
    radius=jnp.array([0.030, 0.030, 0.030, 0.030, 0.030]),
    density=jnp.array([1210.0, 1104.0, 1210.0, 1104.0, 1210.0]),
    young_modulus=jnp.array([1.6464e6] * 5),
    shear_modulus=jnp.array([0.5488e6] * 5),
    material_damping_coefficient=jnp.array([1.96e3] * 5),
    reference_strain=jnp.tile(
        jnp.array([0.0, 0.0, 0.0, 1.0, 0.0, 0.0]), 5
    ),
    # Pneumatic-segment order only: the first and second False entries above.
    chamber_inner_radius=jnp.array([6.39e-3, 6.39e-3]),
    chamber_outer_radius=jnp.array([7.79e-3, 7.79e-3]),
    chamber_distance=jnp.array([20e-3, 20e-3]),
    chamber_azimuth_angles=jnp.tile(
        2.0 * jnp.pi * jnp.arange(3) / 3, (2, 1)
    ),
    # Flattened child lengths, grouped by pneumatic segment.
    pcs_segment_lengths=jnp.array([0.095, 0.095, 0.060, 0.060, 0.060]),
)

robot = ISupport(
    params,
    structure=ISupportStructure(
        rigid_segment_selector=rigid_segment_selector,
        pcs_segment_counts=(2, 3),
    ),
)
```

A non-alternating layout uses the same API:

```python
# pneumatic -> pneumatic -> rigid -> rigid -> pneumatic
structure = ISupportStructure(
    rigid_segment_selector=(False, False, True, True, False),
    pcs_segment_counts=(1, 2, 1),
)
```

If `rigid_segment_selector` is omitted, a five-segment parameter set derives `(True, False, True, False, True)`; a four-segment set derives `(True, False, True, False)`.

## Migration guide

This is intentionally a breaking change with no compatibility aliases.

1. Merge the old pneumatic lengths and `rigid_connector_lengths` into one physical-order `length` array.
2. Replace `rigid_connector_selector` with a same-length `rigid_segment_selector`.
3. Expand every standard body field from pneumatic-only entries to one value per physical segment:
   - `radius`
   - `density`
   - `young_modulus`
   - `shear_modulus`
   - material damping
   - `reference_strain`
4. Expand `reference_strain` to `6 * num_physical_segments` entries. Rigid segments must use `[0, 0, 0, 1, 0, 0]`.
5. If supplying a custom `damping_matrix`, migrate it to physical-segment strain coordinates with shape `(6N, 6N)`.
6. Leave chamber arrays, `pcs_segment_counts`, pressure vectors, and `pcs_segment_lengths` grouped by pneumatic segments only, in the order of the `False` selector entries.
7. Reconstruct `ISupport` when segment types change; the selector is structural/static.

Before:

```python
params = ISupportParams(
    length=jnp.array([0.190, 0.180]),          # pneumatic only
    radius=jnp.array([0.030, 0.030]),
    density=jnp.array([1104.0, 1104.0]),
    rigid_connector_lengths=jnp.array([0.041, 0.027, 0.006]),
    # ...
)
structure = ISupportStructure(
    rigid_connector_selector=(True, True, True),
    pcs_segment_counts=(2, 3),
)
```

After:

```python
params = ISupportParams(
    length=jnp.array([0.041, 0.190, 0.027, 0.180, 0.006]),
    radius=jnp.array([0.030, 0.030, 0.030, 0.030, 0.030]),
    density=jnp.array([1210.0, 1104.0, 1210.0, 1104.0, 1210.0]),
    # Supply all other standard body fields in the same physical order.
    # Chamber and subdivision fields still have two pneumatic groups.
    # ...
)
structure = ISupportStructure(
    rigid_segment_selector=(True, False, True, False, True),
    pcs_segment_counts=(2, 3),
)
```

## Validation

- Full test suite: 1093 passed.
- Post-parameter-update targeted I-SUPPORT system, length, and renderer tests: 45 passed.
- Ruff formatting and lint checks pass for the updated simulation example.

### CODE REVIEW

**Review by Elektron97 — APPROVED** [review]
Everything good!
